.github/workflows/main.yml

```diff
@@ -41,8 +55,33 @@ jobs:
       # Only check cargo lib for now
       - run: cargo clippy -p cargo --lib -- -D warnings
 
+  # Generate strategy matrix for different platforms and channels
+  # (see ci/matrix.json)
+  matrix:
+    runs-on: ubuntu-latest
+    needs:
+    - channel
+    outputs:
+      matrix: ${{ steps.matrix.outputs.matrix }}
+    steps:
+    - uses: actions/checkout@v3
+    - name: Generate strategy matrix
+      id: matrix
+      run: |
+        CHANNEL=${{ needs.channel.outputs.CHANNEL }}
+        MATRIX=$(
+          jq --arg C "$CHANNEL" 'map (. | 
+            if ($C == "beta") then select(.rust | startswith("nightly") | not)
+            elif ($C == "stable") then select(.rust | startswith("stable"))
+            else . end)' ci/matrix.json
+        )
```

Comment:
Can this have a comment explaining roughly what it is doing? It's not too hard to follow, but just a high-level description could help.

Reply:
Added. Let me know if it is clear enough.